### PR TITLE
backend/b2: Switch to official library `Backblaze/blazer`

### LINKED ISSUE #4148: Switch to `Backblaze/blazer` for the B2 backend
According to https://github.com/kurin/blazer/pull/84#issuecomment-1382526369 the new upstream is https://github.com/Backblaze/blazer . We have to wait until https://github.com/Backblaze/blazer/pull/2 is merged before we can switch.

### PR DESCRIPTION
**Pull request by adamantike**
What does this PR change? What problem does it solve?
-----------------------------------------------------

According to https://github.com/kurin/blazer Readme, the repository is no longer maintained, and the official one at
https://github.com/Backblaze/blazer took over.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

Closes #4148

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,7 @@ require (
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.7.0
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.0
 	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v1.1.0
+	github.com/Backblaze/blazer v0.6.1
```
Comment (MichaelEischer) [inline go.mod:8]: Please revert all changes to go.mod _except_ for the blazer library update. Updating the other library belongs into a separate PR.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
See above

File: go.mod (hunk 1 above)
Comment (adamantike) [reply go.mod:8]: Is this desired? The changes I included here were just the results of a `go mod tidy`. The other library updates are because of `blazer` now including a [`go.mod` file](https://github.com/Backblaze/blazer/blob/efa10c0fffe98ce85c0a927a13249cc695a4f8e5/go.mod), and requiring those newer versions.

Comment (MichaelEischer) [reply go.mod:8]: Oh indeed you're right. I'd have prefer to update those dependencies separately, but we can also update them right now.

**Review by MichaelEischer — APPROVED** [review]
LGTM.

**Comment by MichaelEischer** [comment]
(rebased to fix the merge conflict)
